rs/nns/governance/src/proposals/mod.rs

```diff
@@ -17,6 +23,148 @@ pub mod register_known_neuron;
+    /// Computes a topic to a given proposal action at the creation time.
+    pub fn compute_topic_at_creation(&self) -> Result<Topic, GovernanceError> {
+        let topic = match self {
+            ValidProposalAction::ManageNeuron(_) => Topic::NeuronManagement,
+            ValidProposalAction::ManageNetworkEconomics(_) => Topic::NetworkEconomics,
+            ValidProposalAction::Motion(_) => Topic::Governance,
+            ValidProposalAction::ExecuteNnsFunction(execute_nns_function) => execute_nns_function
+                .nns_function()
+                .compute_topic_at_creation()?,
+            ValidProposalAction::ApproveGenesisKyc(_) => Topic::Kyc,
+            ValidProposalAction::AddOrRemoveNodeProvider(_) => Topic::ParticipantManagement,
+            ValidProposalAction::RewardNodeProvider(_) => Topic::NodeProviderRewards,
+            ValidProposalAction::RewardNodeProviders(_) => Topic::NodeProviderRewards,
+            ValidProposalAction::RegisterKnownNeuron(_) => Topic::Governance,
+            ValidProposalAction::DeregisterKnownNeuron(_) => Topic::Governance,
+            ValidProposalAction::CreateServiceNervousSystem(_) => Topic::SnsAndCommunityFund,
+            ValidProposalAction::InstallCode(install_code) => install_code.valid_topic()?,
+            ValidProposalAction::StopOrStartCanister(stop_or_start) => {
+                stop_or_start.valid_topic()?
+            }
+            ValidProposalAction::UpdateCanisterSettings(update_settings) => {
+                update_settings.valid_topic()?
+            }
+            ValidProposalAction::FulfillSubnetRentalRequest(_) => Topic::SubnetRental,
+        };
+        Ok(topic)
+    }
+
+    /// Returns whether proposals with such an action should be allowed to
+    /// be submitted when the heap growth potential is low.
+    pub fn allowed_when_resources_are_low(&self) -> bool {
+        match self {
+            ValidProposalAction::ExecuteNnsFunction(execute_nns_function) => {
+                crate::pb::v1::NnsFunction::try_from(execute_nns_function.nns_function)
+                    .ok()
+                    .is_some_and(|f| f.allowed_when_resources_are_low())
+            }
+            ValidProposalAction::InstallCode(install_code) => {
+                install_code.allowed_when_resources_are_low()
+            }
+            ValidProposalAction::UpdateCanisterSettings(update_settings) => {
+                update_settings.allowed_when_resources_are_low()
+            }
+            _ => false,
+        }
+    }
+
+    /// Returns the ManageNeuron action if this is a ManageNeuron proposal.
+    pub fn manage_neuron(&self) -> Option<&ManageNeuron> {
+        if let ValidProposalAction::ManageNeuron(manage_neuron) = self {
+            Some(manage_neuron)
+        } else {
+            None
+        }
+    }
+
+    /// Returns true if this is a ManageNeuron proposal.
+    pub fn is_manage_neuron(&self) -> bool {
+        self.manage_neuron().is_some()
+    }
```

Comment:
I don't think these "carry their weight", have sufficient "power to weight ratio" to justify their existence. OTOH, I don't hate them; therefore, do as you see fit.